src/runtime-rs/src/shim/src/delete.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use protobuf::Message;
+use shim_proto::shim::DeleteResponse;
+use virtcontainers::Sandbox;
+
+use crate::{Error, Result, ShimExecutor};
+
+impl ShimExecutor {
+    pub fn delete(&mut self) {
+        if let Err(e) = self.args.validate(true) {
+            eprintln!("delete shim err: {}", e);
+            return;
+        }
+
+        match self.do_cleanup() {
+            Ok(rsp) => {
+                rsp.write_to_writer(&mut std::io::stdout())
+                    .expect("failed to write stdout");
```

Comment:
Please can we remove all unecessary `expect()` and `unwrap()` calls that are not listed in the table here:

- https://github.com/kata-containers/kata-containers/blob/main/docs/code-pr-advice.md#unsafe-code

That might require changing interfaces slightly to return a `Result`.

Reply:
Removed uncessary `unwrap()`s in the first revision. But I keep using `expect` for `Mutex.lock()`. On one hand it is most unlikely to return error when acquiring a lock, on the other hand if getting error when acquiring a lock, it would means something panicked on other threads and not recoverable. Then maybe the only correct way is to panic too.